types/lib/elements/dom-repeat.d.ts

```diff
@@ -0,0 +1,268 @@
+     * function should match the sort function passed to `Array.sort`.
+     * Using a sort function has no effect on the underlying `items` array.
+     */
+    sort: Function|null|undefined;
+
+    /**
+     * A function that can be used to filter items out of the view.  This
+     * property should either be provided as a string, indicating a method
+     * name on the element's host, or else be an actual function.  The
+     * function should match the sort function passed to `Array.filter`.
+     * Using a filter function has no effect on the underlying `items` array.
+     */
+    filter: Function|null|undefined;
+
+    /**
+     * When using a `filter` or `sort` function, the `observe` property
+     * should be set to a space-separated list of the names of item
+     * sub-fields that should trigger a re-sort or re-filter when changed.
+     * These should generally be fields of `item` that the sort or filter
+     * function depends on.
+     */
+    observe: string|null|undefined;
+
+    /**
+     * When using a `filter` or `sort` function, the `delay` property
+     * determines a debounce time in ms after a change to observed item
+     * properties that must pass before the filter or sort is re-run.
+     * This is useful in rate-limiting shuffling of the view when
+     * item changes may be frequent.
+     */
+    delay: number|null|undefined;
+
+    /**
+     * Count of currently rendered items after `filter` (if any) has been applied.
+     * If "chunking mode" is enabled, `renderedItemCount` is updated each time a
+     * set of template instances is rendered.
+     */
+    renderedItemCount: number|null|undefined;
+
+    /**
+     * Defines an initial count of template instances to render after setting
+     * the `items` array, before the next paint, and puts the `dom-repeat`
+     * into "chunking mode".  The remaining items will be created and rendered
+     * incrementally at each animation frame therof until all instances have
+     * been rendered.
+     */
+    initialCount: number|null|undefined;
+
+    /**
+     * When `initialCount` is used, this property defines a frame rate (in
+     * fps) to target by throttling the number of instances rendered each
+     * frame to not exceed the budget for the target frame rate.  The
+     * framerate is effectively the number of `requestAnimationFrame`s that
+     * it tries to allow to actually fire in a given second. It does this
+     * by measuring the time between `rAF`s and continuously adjusting the
+     * number of items created each `rAF` to maintain the target framerate.
+     * Setting this to a higher number allows lower latency and higher
+     * throughput for event handlers and other tasks, but results in a
+     * longer time for the remaining items to complete rendering.
+     */
+    targetFramerate: number|null|undefined;
+    _targetFrameTime: number|null|undefined;
+    disconnectedCallback(): any;
```

Comment:
I think these should be typed as returning `void`

Reply:
Fixed in https://github.com/Polymer/polymer/pull/4998